mlflow/store/artifact/databricks_sdk_models_artifact_repo.py

```diff
@@ -0,0 +1,86 @@
+import posixpath
+
+from mlflow.entities import FileInfo
+from mlflow.store.artifact.cloud_artifact_repo import CloudArtifactRepository
+
+DOWNLOAD_CHUNK_SIZE = 1024*1024*1024
+
+def _get_databricks_workspace_client():
+    from databricks.sdk import WorkspaceClient
+
+    return WorkspaceClient()
+
+
+class DatabricksSDKModelsArtifactRepository(CloudArtifactRepository):
+    """
+    Stores and retrieves model artifacts via Databricks SDK, agnostic to the underlying cloud
+    that stores the model artifacts.
+    """
+
+    def __init__(self, model_name, model_version):
+        self.model_name = model_name
+        self.model_version = model_version
+        self.model_base_path = f"/Models/{model_name.replace('.', '/')}/{model_version}"
+        self.w = _get_databricks_workspace_client()
+        super().__init__(self.model_base_path)
+
+    def list_artifacts(self, path=None):
+        dest_path = self.model_base_path
+        if path:
+            dest_path = posixpath.join(dest_path, path)
+
+        file_infos = []
+        resp = self.w.files.list_directory_contents(dest_path)
```

Comment:
`list_directory_contents` result is paginated; can we handle pagination in the response to ensure we retrieve all files in the directory?

Reply:
from the sdk documentations, it looks like the page token and page size are both optional, this alone should get us an iterator to exhaust the full list of directory entries

Reply:
https://databricks-sdk-py.readthedocs.io/en/latest/workspace/files/files.html#databricks.sdk.service.files.FilesAPI.list_directory_contents says:

> We recommend not to set this value unless you are intentionally listing less than the complete directory contents. If unspecified, at most 1000 directory entries will be returned. The maximum value is 1000. Values above 1000 will be coerced to 1000.

Reply:
My understanding is, e.g if there's 1002 entries
The iterator will first fetch 1000 entries,
when `for directory_entry in resp:` exhaust that 1000 entries, the iterator `resp` will have to talk to the Files API to fetch the next page with the remaining 2 entries

Reply:
let me read their impl and get back to you, maybe my assumption was wrong (but it'd be a bit bizzare that they provide a files listing API, and if optional parameters are not provided then it return incorrect results).

Reply:
ah.. i think you're right
`To list all of the entries in a directory, it is necessary to continue requesting pages of entries until the response contains no next_page_token. `

Reply:
it's kinda weird that the API doesn't return a page_token.. it only returned an iterator..

Reply:
ok.. so it seems like my original assumption is correct:
https://github.com/databricks/databricks-sdk-py/blob/main/databricks/sdk/service/files.py#L896-L945
This code takes care of all the page_token logic and makes sure to exhaust all entries out of a directory.
I don't really understand why they would still want to expose the page_size and page_token there..

Reply:
Sounds good. looks like the doc is incorrect.